lib/propolis/src/exits.rs

```diff
@@ -93,11 +93,27 @@ impl From<&bhyve_api::vm_exit_vmx> for VmxDetail {
     }
 }
 
-#[derive(Copy, Clone, Debug)]
+#[derive(Copy, Clone)]
 pub struct InstEmul {
     pub inst_data: [u8; 15],
     pub len: u8,
 }
+
+impl std::fmt::Debug for InstEmul {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        #[cfg(not(feature = "dump-guest-state"))]
+        let inst_data = "<dump-guest-state disabled>";
+
+        #[cfg(feature = "dump-guest-state")]
+        let inst_data = &self.inst_data;
```

Comment:
Rather than gating this output on a feature, perhaps just censor it on the propolis-server side, depending on the configuration there?

Reply:
I considered that but didn't like the ergonomics of it; it would mean that everyone in server who might want to debug-print a `VmExitKind` needs to remember to check what variant it is first to avoid printing an `InstEmul`. We could force that to happen by declining to derive Debug for VmExitKind, but that would put a big `match` into functions like `unhandled_vm_exit` to deal with the one variant that doesn't have a Debug impl, which I also don't love the look of. (I don't think there are any other call sites that would be affected, fortunately.)

Reply:
Following up on this--the latest bits should be much cleaner now; they get rid of all the new propolis-lib feature checks in favor of having a flag that controls whether things tagged as "came from the guest" get redacted, and propolis-server uses the `omicron-build` compile time flag to decide how to set that value.